integration/kubernetes/confidential/sev.bats

```diff
@@ -0,0 +1,323 @@
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+run_kbs() {
+  KBS_DIR="$(mktemp -d /tmp/kbs.XXXXXXXX)"
+  pushd "${KBS_DIR}"
+
+  # Clone and run
+  git clone https://github.com/confidential-containers/simple-kbs.git --branch main
+  (cd simple-kbs && git checkout -b branch_0.1.1 0.1.1)
+  (cd simple-kbs && esudo docker-compose up -d)
+  sleep 5
+
+  # Set KBS_DB_HOST to kbs db container IP
+  KBS_DB_HOST=$(esudo docker network inspect simple-kbs_default \
+    | jq -r '.[].Containers[] | select(.Name | test("simple-kbs_db.*")).IPv4Address' \
+    | sed "s|/.*$||g")
+
+  popd
+}
+
+pull_encrypted_image_and_set_keys() {
+  # Pull encrypted docker image - test workload
+  esudo docker pull quay.io/kata-containers/encrypted-image-tests:encrypted
+
+  # Get encryption key from docker image label
+  ENCRYPTION_KEY=$(esudo docker inspect quay.io/kata-containers/encrypted-image-tests:encrypted \
+    | jq -r '.[0].Config.Labels.enc_key')
+
+  # Get ssh key from docker image label and save to file
+  esudo docker inspect quay.io/kata-containers/encrypted-image-tests:encrypted \
+    | jq -r '.[0].Config.Labels.ssh_key' \
+    | sed "s|\(-----BEGIN OPENSSH PRIVATE KEY-----\)|\1\n|g" \
+    | sed "s|\(-----END OPENSSH PRIVATE KEY-----\)|\n\1|g" \
+    > ${TEST_DIR}/encrypted-image-tests
+
+  # Set permissions on private key file
+  chmod 600 ${TEST_DIR}/encrypted-image-tests
+}
+
+generate_firmware_measurement_with_append() {
+local append_default="tsc=reliable no_timer_check rcupdate.rcu_expedited=1 \
+i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 i8042.noaux=1 noreplace-smp reboot=k \
+cryptomgr.notests net.ifnames=0 pci=lastbus=0 console=hvc0 console=hvc1 quiet panic=1 \
+nr_cpus=1 scsi_mod.scan=none agent.config_file=/etc/agent-config.toml"
+
+  # Gather firmware locations and kernel append for measurement
+  local append=${1:-${append_default}}
+  local kernel_path="$(esudo kata-runtime kata-env --json | jq -r .Kernel.Path)"
+  local initrd_path="$(esudo kata-runtime kata-env --json | jq -r .Initrd.Path)"
+
+  # Generate digest from sev-snp-measure output - this also inserts measurement values inside OVMF image
+  measurement=$(PATH="${PATH}:${HOME}/.local/bin" sev-snp-measure --mode=sev --output-format=base64 \
+    --ovmf="/opt/confidential-containers/share/ovmf/OVMF.fd" \
+    --kernel="${kernel_path}" \
+    --initrd="${initrd_path}" \
+    --append="${append}" \
+  )
+  if [[ -z "${measurement}" ]]; then return 1; fi
+  echo ${measurement}
+}
+
+add_key_to_kbs_db() {
+  measurement=${1}
+
+  # Add key and keyset to DB; If set, add policy with measurement to DB
+  if [ -n "${measurement}" ]; then
+    mysql -u${KBS_DB_USER} -p${KBS_DB_PW} -h ${KBS_DB_HOST} -D ${KBS_DB} <<EOF
+      INSERT INTO secrets VALUES (10, 'key_id1', '${ENCRYPTION_KEY}', 10);
+      INSERT INTO keysets VALUES (10, 'KEYSET-1', '["key_id1"]', 10);
+      INSERT INTO policy VALUES (10, '["${measurement}"]', '[]', 0, 0, '[]', now(), NULL, 1);
+EOF
+  else
```

Comment:
@ryansavino for the sake of self-learn: if the image is unencrypted then it doesn't need to validate the measurement?

Reply:
For now the measurement is still done for the unencrypted image. @fitzthum and I will look into improving the unencrypted workflow.

Reply:
Yeah there is a weird dependency thing here. I will make an issue for it soon and hopefully we will fix for next release by supporting blank keysets in the kata-config.